# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserHost.h
# Commit: 810b73719871
# Full Hash: 810b7371987139844429d0206f9da6a7701a1efc
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 04:41:59
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 13 - Stop inheriting nsIRemoteTab interface in BrowserParent. r=nika
#   
#   This commit removes nsIRemoteTab as a parent class from BrowserParent,
#   so that BrowserHost is the only concrete implementation of nsIRemoteTab.
#   
# ==============================================================================

diff -r ee10a8254481 -r 810b73719871 dom/ipc/BrowserHost.h
--- a/dom/ipc/BrowserHost.h	Wed May 08 11:45:23 2019 -0500
+++ b/dom/ipc/BrowserHost.h	Wed May 08 14:34:47 2019 -0500
@@ -11,9 +11,18 @@
 #include "mozilla/dom/RemoteBrowser.h"
 #include "mozilla/dom/BrowserParent.h"
 
+class nsPIDOMWindowOuter;
+
 namespace mozilla {
+
+namespace a11y {
+class DocAccessibleParent;
+}  // namespace a11y
+
 namespace dom {
 
+class Element;
+
 /**
  * BrowserHost manages a remote browser from the chrome process.
  *
@@ -29,6 +38,8 @@
 
   explicit BrowserHost(BrowserParent* aParent);
 
+  static BrowserHost* GetFrom(nsIRemoteTab* aRemoteTab);
+
   NS_DECL_ISUPPORTS
   // nsIRemoteTab
   NS_DECL_NSIREMOTETAB
@@ -44,6 +55,12 @@
   BrowsingContext* GetBrowsingContext() const override;
   nsILoadContext* GetLoadContext() const override;
 
+  Element* GetOwnerElement() const { return mRoot->GetOwnerElement(); }
+  already_AddRefed<nsPIDOMWindowOuter> GetParentWindowOuter() const {
+    return mRoot->GetParentWindowOuter();
+  }
+  a11y::DocAccessibleParent* GetTopLevelDocAccessible() const;
+
   void LoadURL(nsIURI* aURI) override;
   void ResumeLoad(uint64_t aPendingSwitchId) override;
   void DestroyStart() override;